transport/http2_client.go

```diff
@@ -318,6 +329,9 @@ func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (_ *Strea
 func (t *http2Client) CloseStream(s *Stream, err error) {
 	t.mu.Lock()
 	delete(t.activeStreams, s.id)
+	if t.streamsQuota != nil {
+		t.streamsQuota.add(1)
```

Comment:
The locking model here still seems inconsistent - the streamsQuota.add call here is with t.mu locked, but at 255 t.mu is still unlocked.  (Is the streamsQuota op atomic?  If so, could this be moved after t.mu.Unlock()?)